core/embed/rust/src/ui/flow/flow.rs

```diff
@@ -0,0 +1,202 @@
+use crate::{
+    error,
+    time::{Duration, Instant},
+    ui::{
+        animation::Animation,
+        component::{Component, Event, EventCtx},
+        flow::{base::Decision, swipe::Swipe, FlowMsg, FlowState, FlowStore, SwipeDirection},
+        geometry::{Offset, Rect},
+        shape::Renderer,
+    },
+};
+
+const ANIMATION_DURATION: Duration = Duration::from_millis(333);
+
+/// Given a state enum and a corresponding FlowStore, create a Component that
+/// implements a swipe navigation between the states with animated transitions.
+///
+/// If a swipe is detected:
+/// - currently active component is asked to handle the event,
+/// - if it can't then FlowState::handle_swipe is consulted.
+pub struct Flow<Q, S> {
```

Comment:
renamed `Flow` to `SwipeFlow` in c741b2baba430ed3caad803c52c333fae9b7e9c4 since it's actually dependent on swiping
`FlowStore` is generic so I left it as it is